lib/spack/docs/tutorial_configuration.rst

```diff
@@ -0,0 +1,700 @@
+      ^zlib@1.2.11%clang@3.8.0-2ubuntu4+pic+shared arch=linux-ubuntu16.04-x86_64
+
+--------------------------------
+Variant Preference Configuration
+--------------------------------
+
+The packages configuration file can also set variant preferences for
+packages. For example, let's change our preferences to build all
+packages without static libraries. We will accomplish this by turning
+off the `shared` variant on all packages that have one.
+
+.. code-block:: yaml
+
+  packages:
+    all:
+      compiler: [clang, gcc, intel, pgi, xl, nag]
+      providers:
+        mpi: [mpich, openmpi]
+      variants: ~shared
+
+We can check the effect of this command with `spack spec hdf5` again.
+
+.. code-block:: console
+
+  $ spack spec hdf5
+  Input spec
+  --------------------------------
+  hdf5
+
+  Normalized
+  --------------------------------
+  hdf5
+      ^zlib@1.1.2:
+
+  Concretized
+  --------------------------------
+  hdf5@1.10.1%clang@3.8.0-2ubuntu4+cxx~debug~fortran~hl+mpi+pic~shared~szip~threadsafe arch=linux-ubuntu16.04-x86_64
+      ^mpich@3.2%clang@3.8.0-2ubuntu4 device=ch3 +hydra netmod=tcp +pmi+romio~verbs arch=linux-ubuntu16.04-x86_64
+      ^zlib@1.2.11%clang@3.8.0-2ubuntu4+pic~shared arch=linux-ubuntu16.04-x86_64
+
+So far we have only made global changes to the package preferences. As
+we've seen throughout this tutorial, hdf5 builds with MPI enabled by
+default in Spack. If we were working on a project that would routinely
+need serial hdf5, that might get annoying quickly, having to type
+`hdf5~mpi` all the time. Instead, we'll update our preferences for
+hdf5.
+
+.. code-block:: yaml
+
+  packages:
+    all:
+      compiler: [clang, gcc, intel, pgi, xl, nag]
+      providers:
+        mpi: [mpich, openmpi]
+      variants: ~shared
+    hdf5:
+      variants: ~mpi
+
+Now hdf5 will concretize without an MPI dependency by default.
+
+.. code-block:: console
+
+  $ spack spec hdf5
+  Input spec
+  --------------------------------
+  hdf5
+
+  Normalized
+  --------------------------------
+  hdf5
+      ^zlib@1.1.2:
+
+  Concretized
+  --------------------------------
+  hdf5@1.10.1%clang@3.8.0-2ubuntu4+cxx~debug~fortran~hl+mpi+pic~shared~szip~threadsafe arch=linux-ubuntu16.04-x86_64
+      ^zlib@1.2.11%clang@3.8.0-2ubuntu4+pic~shared arch=linux-ubuntu16.04-x86_64
+
+In general, every attribute that we can set for all packages we can
+set separately for an individual package.
+
+--------------------------
+External Packages in Spack
+--------------------------
+
+The packages configuration file also controls when Spack will build
+against an externally installed package. On these systems we have a
+pre-installed zlib.
+
+.. code-block:: yaml
+
+  packages:
+    all:
+      compiler: [clang, gcc, intel, pgi, xl, nag]
+      providers:
+        mpi: [mpich, openmpi]
+      variants: ~shared
+    hdf5:
+      variants: ~mpi
+    zlib:
+      paths:
+        zlib@1.2.11%gcc@5.4.0+pic+shared arch=linux-ubuntu16.04-x86_64: /usr
```

Comment:
I think it would be better to give an older version of `zlib`, as that is more likely to be installed on the system. This will also make the difference between what we want Spack to use and what Spack actually uses more clear. Also, users who add external packages rarely know exactly how they were built. I never add anything other than the name of the package and the version.